lib/skate_web/controllers/detours_controller.ex

```diff
@@ -64,12 +69,16 @@ defmodule SkateWeb.DetoursController do
           connection_end_location
         )
 
+      {:ok, ors_result} =
+        OpenRouteServiceAPI.directions([connection_start] ++ waypoints ++ [connection_end])
+
```

Comment:
issue: this assertion now means that any successful computation of missed stops and the route segments will be overshadowed by ORS errors, which I don't think is a great UX

(I put a waypoint in the ocean in this example)
![image](https://github.com/mbta/skate/assets/6913198/74b50e49-caca-46d3-aa75-f4cbfe03ae39)

I don't think we should require here that directions are successful to be able to compute the other things.

Reply:
I'm thinking of this as a follow-up, but would it be possible for the state machine to disallow placing more waypoints/endpoint/etc when there's a routing error? Because if there's a routing error for waypoints `1` thru `n`, then we already know that there'll be a routing error for waypoints `1` thru `n+1`.

(We probably only want to have this behavior on routing error, not on other errors)

That feels like something the state machine would be able to do pretty easily.

Reply:
> but would it be possible for the state machine to disallow placing more waypoints/endpoint/etc when there's a routing error?

Sounds easily doable to me! Happy to find some time to pair on it if you'd like.

Reply:
Do you want ☝️ to be a part of this PR? Or should we pair on it as a follow-up?

Reply:
Lets do it as a follow-up 👍🏻

Reply:
https://app.asana.com/0/1148853526253420/1207750816978131/f

Reply:
Take 2: https://app.asana.com/0/1148853526253420/1207768837488463/f